playpen/all_27.sh

```diff
@@ -1,22 +1,64 @@
 #!/bin/bash
+# Downgrade a Pulp system from master to 2.7.
+#
+# WARNING: This script enabled and disables system-wide yum/dnf repositories and
+# performs other system-wide actions. Execute with care.
+set -euo pipefail
 
-#  When using this script, you will need to fix your database accordingly. If moving backward, you
-#  will need to drop the database, and you will always need to run the migrations.
+# This associative array is incomplete. Patches welcome.
+declare -rA repos_branches=(
+    [pulp]=2.7-dev
+    [pulp_ostree]=1.0-dev
+    [pulp_puppet]=2.7-dev
+    [pulp_python]=1.0-dev
+    [pulp_rpm]=2.7-dev
+)
 
+# Reinstall plugins we know how to handle, and uninstall all others.
+pushd ~/devel
+for repo in pulp*; do
+    # Defend against e.g. a file named "pulp_log.txt".
+    if [ ! -d "$repo" ]; then
+        continue
+    fi
 
-# pulp_docker, pulp_python, pulp_ostree excluded because they are versioned differently
-for repo in pulp pulp_puppet pulp_rpm
-do
-    pushd /home/vagrant/devel/$repo
-    git checkout 2.7-dev
-    find ~/devel -name "*.py[c0]" -delete
-    sudo ./pulp-dev.py --install
+    pushd "$repo"
+    if [ -n "${repos_branches[$repo]:-}" ]; then
+        git checkout "${repos_branches[$repo]}"
+        find . -name '*.py[c0]' -delete
+        sudo ./pulp-dev.py --install
+    else
+        sudo ./pulp-dev.py --uninstall
+    fi
     popd
 done
+popd
 
-for repo in pulp_docker pulp_python
-do
-    pushd /home/vagrant/devel/$repo
-    sudo ./pulp-dev.py --uninstall
-    popd
-done
+set -x
+
+# These aren't used by Pulp 2.7.
+sudo rm /etc/httpd/conf.d/pulp_content.conf
```

Comment:
Does it hurt anything to leave these? The devs who might use this script might "pop over" to 2.7 to see if a bug was present there, and then go right back to master where they would be surprised that their config files have been restored to defaults.